src/metrics/connection.rs

```diff
@@ -2,20 +2,28 @@ use goodmetrics::SumHandle;
 
 pub struct ConnectionGuard {
     connections_closed: SumHandle,
+    total_active_connections: SumHandle,
 }
 
 impl ConnectionGuard {
-    /// Creates a new `ConnectionGuard` instance.
-    /// This instance will increment the `connections_opened` counter
-    pub fn new(connections_opened: SumHandle, connections_closed: SumHandle) -> Self {
+    pub fn new(
+        connections_opened: SumHandle,
+        connections_closed: SumHandle,
+        total_active_connections: SumHandle,
+    ) -> Self {
         connections_opened.observe(1);
-        Self { connections_closed }
+        total_active_connections.observe(1);
+        Self {
+            connections_closed,
+            total_active_connections,
+        }
     }
 }
 
 impl Drop for ConnectionGuard {
     fn drop(&mut self) {
         // When the guard is dropped, we assume the connection is closed.
         self.connections_closed.observe(1);
+        self.total_active_connections.observe(-1);
```

Comment:
these gauges reset; I think this is not exactly what you're looking for. The +1/-1 only works when your graphing aggregation window is wide enough to capture the +'s and -'s. I think you actually want to report the snapshot of actual connection count.

this is a thing goodmetrics should provide a tool for, but in the interim (or unless you want to pr and add a tool) you can do +1/-1 on an Arc<AtomicUsize> and spawn a task that `total_active_connections.observe(connections.load())`s every couple secconds. IMO you'd be best served with a statisticsethandle, of the options you currently have.